### ISSUE TITLE
Introduce a build option to disable mape and it's required GTK dependencies

### ISSUE THREAD

**Issue by tgurr** [openclonk/openclonk#75]
You've removed the USE_GTK https://github.com/openclonk/openclonk/commit/b7359e0c27210b59969b5d05b184f04683cdf16c option, however the checks for gtk stuff are still there and are currently automagic. As they're still required for the mape editor.

Having an editor/mape CMake option with all the GTK stuff behind would allow us to introduce an editor option to our downstream package which would reduce the required dependencies for users only wanting to play the game.

**Comment by lluchs** [comment]
Yes, the current system where features are turned on/off implicitly depending on whether dependencies are available is not very good. It makes people building manually miss out on features like the editor and it makes packaging harder because package managers often don't sandbox the building process.

We should add explicit options for most things that needs additional dependencies and have the build fail if these aren't available. This includes:

 - mape (needs GTK)
 - the editor (needs Qt)
 - sound (needs OpenAL)
 - maybe UPnP (needs miniupnpc)

Enabling mape should then also include mape in the `install` target. Similarly, we're currently always building openclonk-server, but don't install it. Not sure what to do there. We already break it accidentally often enough, disabling it per default would make this worse.

**Comment by jcaesar** [comment]
The experimental [meson build](https://github.com/jcaesar/openclonk/blob/meson/meson.build) I made does this, mostly… Now you have me wonder whether I should continue that project.
